### DISCUSSION TITLE
Sending single packet without flushing

### DISCUSSION THREAD

**Discussion by andreasdc** [netty/netty#12451] [Q&A]
Hey, I'm trying to make an addition to my app, I have a lot of packets in the flush queue, but I suppose when there are more packets it needs more time to send. I want to send single packet in the hot code, is it possible?

**Comment by SplotyCode** [comment]
Netty does not have this functionality. And i am afraid it would be quite hard to implement and highly unstable/dependent on the channel. For some implementations Channel#doWrite() might be enough (look at DefaultChannelPipeline.HeadContext and Channel.Unsafe).
But what you could do is to not really use the netty flush queue (directly call flush) and write somethinkglike the ChannelOutboundBuffer yourself. It could be as simple as an EvenLoop Task that flushes all messages in its own buffer every few millis.

**Reply by andreasdc** [reply]
> Netty does not have this functionality. And i am afraid it would be quite hard to implement and highly unstable/dependent on the channel. For some implementations Channel#doWrite() might be enough (look at DefaultChannelPipeline.HeadContext and Channel.Unsafe). But what you could do is to not really use the netty flush queue (directly call flush) and write somethinkglike the ChannelOutboundBuffer yourself. It could be as simple as an EvenLoop Task that flushes all messages in its own buffer every few millis.

But in that way it will wait for the flush or flush with all queued packets, right?
